Framework/Geometry/test/RasterizeTest.h

```diff
@@ -0,0 +1,176 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_GEOMETRY_RASTERIZETEST_H_
+#define MANTID_GEOMETRY_RASTERIZETEST_H_
+
+#include "MantidGeometry/Objects/CSGObject.h"
+#include "MantidGeometry/Objects/IObject.h"
+#include "MantidGeometry/Objects/ShapeFactory.h"
+#include "MantidGeometry/Rasterize.h"
+#include "MantidKernel/V3D.h"
+#include <cxxtest/TestSuite.h>
+#include <numeric>
+
+using namespace Mantid::Geometry;
+using Mantid::Kernel::V3D;
+
+class RasterizeTest : public CxxTest::TestSuite {
+private:
+  // all examples are taken from ShapeFactoryTest with some small modifications
+  const std::string cylinderXml =
+      "<cylinder id=\"shape\"> "
+      "<centre-of-bottom-base x=\"0.0\" y=\"-1.5\" z=\"0\"/>"
+      "<axis x=\"0.0\" y=\"1\" z=\"0\" /> "
+      "<radius val=\"0.1\" /> "
+      "<height val=\"3\" /> "
+      "</cylinder>";
+  const std::string cylinderOffsetXml =
+      "<cylinder id=\"shape\"> "
+      "<centre-of-bottom-base x=\"0.0\" y=\"3.0\" z=\"0.0\"/>"
+      "<axis x=\"0.0\" y=\"0.0\" z=\"1\" /> "
+      "<radius val=\"0.1\" /> "
+      "<height val=\"3\" /> "
+      "</cylinder>";
+  static constexpr double CYLINDER_VOLUME = M_PI * .1 * .1 * 3.;
+
+  const std::string sphereXml = "<sphere id=\"shape\"> "
+                                "<centre x=\"0.0\"  y=\"0.0\" z=\"0.0\" /> "
+                                "<radius val=\"3.2\" /> "
+                                "</sphere>"
+                                "<algebra val=\"shape\" /> ";
+  const std::string sphereOffsetXml = "<sphere id=\"shape\"> "
+                                      "<centre x=\"4\"  y=\"4\" z=\"4\" /> "
+                                      "<radius val=\"3.2\" /> "
+                                      "</sphere>"
+                                      "<algebra val=\"shape\" /> ";
+  static constexpr double SPHERE_VOLUME = (4. / 3.) * M_PI * 3.2 * 3.2 * 3.2;
+
+  boost::shared_ptr<IObject> createCylinder(bool centered) {
+    if (centered)
+      return ShapeFactory().createShape(cylinderXml);
```

Comment:
We do have some shape-related functions in `ComponentCreationHelper`. Did you check those? Could some of this shape creation stuff fit there?